uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveBalancesSelect.php

```diff
@@ -227,15 +223,28 @@ private function getLeaveBalances($contacts) {
    * @return array
    */
   private function hasActiveLeaveManagerCondition() {
-    $today =  '"' . date('Y-m-d') . '"';
-    $leaveApproverRelationshipTypes = $this->getLeaveApproverRelationshipsTypesForWhereIn();
-
     $conditions = [];
-    $conditions[] = 'rt.is_active = 1';
-    $conditions[] = 'rt.id IN(' . implode(',', $leaveApproverRelationshipTypes) . ')';
-    $conditions[] = 'r.is_active = 1';
-    $conditions[] = "(r.start_date IS NULL OR r.start_date <= {$today})";
-    $conditions[] = "(r.end_date IS NULL OR r.end_date >= {$today})";
+
+    if($this->shouldFilterByManager()) {
+      $today =  '"' . date('Y-m-d') . '"';
+      $leaveApproverRelationshipTypes = $this->getLeaveApproverRelationshipsTypesForWhereIn();
+
+      $conditions[] = 'rt.is_active = 1';
+      $conditions[] = 'rt.id IN(' . implode(',', $leaveApproverRelationshipTypes) . ')';
+      $conditions[] = 'r.is_active = 1';
+      $conditions[] = "(r.start_date IS NULL OR r.start_date <= {$today})";
+      $conditions[] = "(r.end_date IS NULL OR r.end_date >= {$today})";
+
+      if(!$this->leaveManagerService->currentUserIsAdmin()) {
+        $managerID = (int)CRM_Core_Session::getLoggedInContactID();
+        $conditions[] = "r.contact_id_b = {$managerID}";
+      }
+
+      if(!empty($this->params['managed_by'])) {
```

Comment:
Thinking  of a scenario where the current user is a manager and managed_by is present which has a value other that the current logged in user ID. Looks like both IF conditions will be met and we have ` r.contact_id_b = {$currentUserID} AND r.contact_id_b = {$managerID}"`, although no results will be returned. 
Wondering why the previous if/else condition is not used again so as to avoid having this condition `$conditions[] = "r.contact_id_b = {$managerID}";` twice in such scenario?

Reply:
The previous version of this had a single `r.contact_id_b = {$managerID}` where the $managerID was given by:

```php
$managerID = null;
if (!currentUserIsAdmin()) {
  $managerID = getLoggedInContactID();
} elseif(!empty($params['managed_by'])) {
  $managerID = $params['managed_by'];
}
```

With that, this is what would happen:
- An admin tries to use the API without passing a manager ID: No join with the relationship tables and all the contacts could be returned
- An admin tries to use the API passing a manager ID: A join with the relationship tables was added and the value passed to the managed_by param was used on the r.contact_id_b condition, meaning that only contacts managed by the given manager would be returned.
- A manager tries to use the API without passing a manager ID: A join with the relationship tables was added and the ID of the current logged in contact would be used as the managerID, meaning that only contacts managed by the current user would be returned
- A manager tries to use the API passing a manager_id of another manager: A join with the relationship tables was added, but the given managed_by value is discarded, as we always use the ID of the current user when they are not admins. This means that the managees of the current user would be returned

On the last scenario above, the correct behavior would be an empty response instead of a list of contacts managed by the current user. That's why we need two separate conditions.

Reply:
Got it.